### DISCUSSION TITLE
Train_MB_Size & eval_mb_size Clarification

### DISCUSSION THREAD

**Discussion by DannyFGitHub** [ContinualAI/avalanche#873] [Q&A] [answered]
# What does the train_mb_size and eval_mb_size do to the number of images?

Say a folder has 1500 images in total for three different classes.
First the task incremental benchmark divides 1500 by the number of experiences. Say 20 experiences.
1500 / 20 = 75
Then for the strategy, although it says that train_mb_size is the train minibatch size, it rather seems to be dividing 75 by train_mb_size?

If so, if I set train_mb_size to 5, when I run the experiment I get 15 per experience.
15 * 20 experiences = 300 images
Original amount is 1500 - 300 = 1200
Does it simply not use the remaining 1200 images?

I struggled to find explanation in documentation e.g. Naive simply says
train_mb_size – The train minibatch size. Defaults to 1.
eval_mb_size – The eval minibatch size. Defaults to 1.
https://avalanche-api.continualai.org/en/stable/generated/avalanche.training.Naive.html?highlight=Naive

To confirm the question is: 
Does the strategy divide number of images per experience by the train_mb_size integer, or is the train_mb_size meant to be the set size for the minibatches? What does it do with remaining images?

**Comment by AntonioCarta** [comment, marked as answer]
`train_mb_size` and `eval_mb_size` are arguments of the data loader. Basically, they iterate over the current dataset, returning mini-batches of the chosen size.
